suite-common/wallet-graph/src/graphThunks.ts

```diff
@@ -5,146 +5,333 @@ import { getBlockbookSafeTime } from '@suite-common/suite-utils';
+        : fallbackMovement;
+};
+
+/**
+ * We need to know what is the account balance at the beginning of selected graph time frame
+ * (to have balance history as a continuous line).
+ * @param account
+ * @param startOfRangeDate
+ */
+const getAccountBalanceAtStartOfRange = async (
+    account: Account,
+    startOfRangeDate: Date,
+): Promise<LineGraphTimeFrameItemAccountBalance> => {
+    const accountBalanceHistoryToStartOfRange = await fetchAccountBalanceHistory(account, {
+        to: getBlockbookSafeTime(getUnixTime(startOfRangeDate)),
+        groupBy: 3600, // day
+    });
+    if (accountBalanceHistoryToStartOfRange?.length) {
+        return accountBalanceHistoryToStartOfRange[accountBalanceHistoryToStartOfRange.length - 1];
+    }
+    // fallback to the oldest if nothing is found
+    const oldestAccountBalanceMovement = await getOldestAccountBalanceMovement(account);
+    return oldestAccountBalanceMovement;
+};
+
+const getMinutesBackToStartOfRange = async (
     accounts: Account[],
     timeFrame: LineGraphTimeFrameValues,
 ): Promise<number> => {
     const { valueBackInMinutes } = timeSwitchItems[timeFrame];
     if (valueBackInMinutes) {
         return valueBackInMinutes;
     }
-    const oldestAccountBalanceChangeUnixTime = await getOldestAccountBalanceMovementTimestamp(
-        accounts,
-    );
+    const oldestAccountBalanceChangeUnixTime =
+        await getOldestAccountBalanceMovementTimestampFromAllAccounts(accounts);
     return differenceInMinutes(new Date(), new Date(oldestAccountBalanceChangeUnixTime * 1000));
 };
 
-const fetchAccountsGraphData = async (
-    accounts: Account[],
-    { from, to, groupBy }: { from: number; to: number; groupBy: number },
+const getFiatRatesForSelectedTimeFrame = async (
+    timeFrameItem: Required<LineGraphTimeFrameConfiguration>,
+    startOfRangeDate: Date,
+    endOfRangeDate: Date,
 ) => {
-    const promises = accounts.map(async account => {
-        const response = await TrezorConnect.blockchainGetAccountBalanceHistory({
-            coin: account.symbol,
-            descriptor: account.descriptor,
-            from,
-            to,
-            groupBy,
-        });
-        try {
-            if (response?.success) {
-                // TODO
-            }
-        } catch (error) {
-            // eslint-disable-next-line no-console
-            console.log(error);
+    const { stepInMinutes } = timeFrameItem;
+
+    const datesInRange = eachMinuteOfInterval(
+        {
+            start: startOfRangeDate.getTime(),
+            end: endOfRangeDate.getTime(),
+        },
+        {
+            step: stepInMinutes,
+        },
+    );
+    const datesInRangeUnixTime = datesInRange.map(date => getBlockbookSafeTime(getUnixTime(date)));
+
+    // FIXME mobile app currently supports only btc so it is hardcoded for now
+    const fiatRatesForDatesInRange = await getFiatRatesForTimestamps(
+        { symbol: 'btc' },
```

Comment:
I am not happy about hardcoding it here, can we pass it as param and hardcode it in place of usage (mobile app)?

Reply:
When I wanted to fix this, it turned out that in the generality of the chart, I had not counted with any other currencies 
 than BTC to be shown in the dashboard overview of the entire portfolio. This required another major extension of functionality... so it is not hardcoded anymore but there is like a bunch of new code... 

https://github.com/trezor/trezor-suite/pull/6585/commits/5102c53270fc8dc52b26d08917704910c2c1d9e2